src/Microsoft.VisualStudio.Threading.Tests/ReentrantSemaphoreTestBase.cs

```diff
@@ -227,23 +228,24 @@ public void ExitInAnyOrder_ExitInAcquisitionOrder()
         this.semaphore = this.CreateSemaphore(ReentrantSemaphore.ReentrancyMode.Freeform);
         this.ExecuteOnDispatcher(async delegate
         {
-            var releaser1 = new AsyncManualResetEvent();
+            var outerReleaser = new AsyncManualResetEvent();
             var innerReleaser = new AsyncManualResetEvent();
             Task innerOperation = null;
-            await this.semaphore.ExecuteAsync(delegate
+            await this.semaphore.ExecuteAsync(async delegate
             {
-                innerOperation = EnterAndUseSemaphoreAsync(innerReleaser);
-                return TplExtensions.CompletedTask;
+                innerOperation = EnterAndUseSemaphoreAsync(innerReleaser, outerReleaser);
+                await outerReleaser;
             });
```

Comment:
Ah, these changes were probably necessary due to the unconditional `await Task.Yield();` we added. Fixing that as I mention at that place in the code should allow you to revert these changes.

Reply:
Is that behavior you want the test to enforce though? That it does not yield at all when there is no semaphore contention?